dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -216,86 +220,110 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                 }
             }
 
-            BackgroundTaskRegistration registeredTaskFromBuilder = nullptr;
+            BackgroundTaskRegistration registeredTaskFromBuilder{ nullptr };
 
             auto scopeExitToCleanRegistrations = wil::scope_exit(
                 [&]()
                 {
-                    if (cookie > 0)
-                    {
-                        LOG_IF_FAILED(::CoRevokeClassObject(cookie));
-                    }
+                    s_comActivatorRegistration.reset();
 
                     // Clean the task registration only if it was created during this call
                     if (registeredTaskFromBuilder)
                     {
-                        registeredTask.Unregister(true);
+                        registeredTaskFromBuilder.Unregister(true);
                     }
                 }
             );
 
-            if (WI_IsFlagSet(registrationOptions, PushNotificationRegistrationOptions::ComActivator))
+            if (WI_IsFlagSet(registrationActivators, PushNotificationRegistrationActivators::ComActivator))
             {
+                {
+                    auto lock = s_activatorInfoLock.lock_exclusive();
+                    THROW_HR_IF_MSG(E_INVALIDARG, s_comActivatorRegistration, "ComActivator already registered.");
+                }
+
                 GetWaitHandleForArgs().create();
 
                 THROW_IF_FAILED(::CoRegisterClassObject(
                     taskClsid,
                     winrt::make<PushNotificationBackgroundTaskFactory>().get(),
                     CLSCTX_LOCAL_SERVER,
                     REGCLS_MULTIPLEUSE,
-                    &cookie));
+                    &s_comActivatorRegistration));
             }
 
             if (builder)
             {
                 registeredTaskFromBuilder = builder.Register();
             }
 
-            PushNotificationRegistrationToken token = { cookie, registeredTaskFromBuilder };
             scopeExitToCleanRegistrations.release();
 
-            PushNotificationTelemetry::ActivatorRegisteredByApi(S_OK, details.Options());
+            auto lock = s_activatorInfoLock.lock_exclusive();
+            s_pushTriggerRegistration = registeredTaskFromBuilder;
 
-            return token;
+            PushNotificationTelemetry::ActivatorRegisteredByApi(S_OK, details.Activators());
         }
 
-        catch (...)
+        catch(...)
         {
             HRESULT hrError = wil::ResultFromCaughtException();
-            PushNotificationTelemetry::ActivatorRegisteredByApi(hrError,
-                details == nullptr ? PushNotificationRegistrationOptions::Undefined : details.Options());
+            PushNotificationTelemetry::ActivatorRegisteredByApi(hrError, 
+                details == nullptr ? PushNotificationRegistrationActivators::Undefined : details.Activators());
 
             THROW_HR(hrError);
```

Comment:
Consider removing this and just saying `throw;` here which propagates the current exception instead of effectively creating a new one.